src/ResourceIDs.hh

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <bitset>
+#include <stdint.h>
+
+constexpr int MIN_RES_ID = -32768;
+constexpr int MAX_RES_ID = 32767;
+
+
+// A set of resource IDs
+class ResourceIDs {
+  public:
+    explicit ResourceIDs(bool all_included = false) : bits() {
+      if (all_included)
+        this->bits.set();
+    }
+    
+    bool operator[](int res_id) const {
+      return this->bits.test(res_id - MIN_RES_ID);
+    }
+    
+    ResourceIDs& operator+=(int res_id) {
+      this->bits.set(res_id - MIN_RES_ID);
+      return *this;
+    }
+    
+    ResourceIDs& operator-=(int res_id) {
+      this->bits.reset(res_id - MIN_RES_ID);
+      return *this;
+    }
+    
+    void reset(bool all_included) {
+      if (all_included) {
+        this->bits.set();
+      } else {
+        this->bits.reset();
+      }
+    }
+    
+  private:
+    std::bitset<MAX_RES_ID - MIN_RES_ID + 1>  bits;
+    
+    friend void fprint(FILE*, const ResourceIDs&, bool);
+};
+
+void fprint(FILE* file, const ResourceIDs& res_ids, bool new_line);
```

Comment:
> This is a good idea - thanks for building it. When it's merged, I might use it in resource_dasm as well to make `--target` more versatile. Requesting changes for some style nitpicks.
> 
> Edit: Missed your comment above :) I'll add it myself in a few days if you don't get around to it.

My idea was:
- `--target-type=<type>:<ids>` limits the disassembled IDs of `<type>` (this would change existing behavior)
- `--target-id=<ids>` limits the disassembled IDs in general
- `--skip=<ids>` excludes the IDs in general

What do you think? Or would the change make `--skip` redundant?

What about `!<id>` to mean -32768..32767 *except* `<id>` (and similarly for ID ranges: `!<min id>..<max id>`) to integrate the functionality of `--skip` into `--target-id`?

Reply:
Fixed.